content/artifacts/artifacts-helper/plugin.js

```diff
@@ -0,0 +1,241 @@
+            <span style=${{ flex: 1 }}></span>
+            <span style=${Spacing}>Cur Blk: ${props.chainBlockNumber < 1 ? "Loading" : props.chainBlockNumber}</span>
+        </div>
+        <div style=${{ display: 'flex' }}>
+            ${numsHTML}
+        </div>
+        ${warningHTML}
+        <div style=${{
+            maxHeight: "300px", overflowX: "hidden", overflowY: "scroll",
+            display: 'grid', gridTemplateColumns: 'repeat(3,1fr)', border: '1px solid gray'
+        }}
+            otherstyle="grid-template-columns:repeat(auto-fill, minmax(4em, max-content));">
+            ${itemsHTML}
+        </div>
+    </div>
+    `;
+}
+
+
+class Plugin {
+    //container: any; // Plugin container, used in render()
+    //pluginHTMLNode: any; // Root node of the plugin panel
+
+    //_chainBlockNumber_Subscriber: any;
+    //chainBlockNumber: number;
+    //chainBlockNumberplanets: any[];
+    container;
+    pluginHTMLNode;
+    planets;
+    chainBlockNumber;
+    _chainBlockNumber_Subscriber;
+
+    constructor() {
+        this._chainBlockNumber_Subscriber = ui.getEthConnection().blockNumber$.subscribe(x => {
+            let t = this.chainBlockNumber;
+            this.chainBlockNumber = x;
+            if (t < 1) {
+                this.refreshPlanetStates();
+            }
+        });
+        this.container = null;
+        this.pluginHTMLNode = null;
+        this.chainBlockNumber = 0;
+        this.planets = [];
+    }
+
+    destroy() {
+        this._chainBlockNumber_Subscriber.unsubscribe();
+        render(null, this.container, this.pluginHTMLNode);
+    }
+
+    getAllRuins() {
+        return df.getMyPlanets().filter((planet) => planet.planetType === PlanetType.RUINS);
+    }
+
+    refreshPlanetStates() {
+        console.log("Refresing Planets with Artifacts...");
+        this.planets = this.getAllRuins().map((p) => [p, checkPlanetType(p, this.chainBlockNumber)]).sort((p, q) => (p[1] - q[1]));
+        this.render(this.container);
+    }
+
+    doProspectAndFind() {
+        const do_find = (planet) => {
+            console.log("Post F on ", planet.locationId);
+            df.findArtifact(planet.locationId);
+            /*.then(
+                            v => {
+                                console.log("Finish Finding    artifact on ", planet.locationId);
+                            },
+                            e => {
+                                console.log("Error  Finding    artifact on ", planet.locationId, " ", e);
+                            });*/
+        };
+        const do_prospect = (planet) => {
+            console.log("Post P on ", planet.locationId);
+            df.prospectPlanet(planet.locationId).then(
+                v => {
+                    do_find(planet);
+                },
+                e => {
+                    console.log("Error Prospecting artifact on ", planet.locationId, " ", e);
+                });
```

Comment:
It looks like this is always caught and never rejects

Reply:
Do you mean I should re-throw `e` ? Is that for showing the error on console?